test/scripts/products/h24-verify-selfmanaged-apicast-and-custom-policy/h24-verify-selfmanaged-apicast-and-custom-policy.go

```diff
@@ -0,0 +1,471 @@
+	token3scale string, admin_portal string, apicastNamespace string) error {
+	log.Info("Create adminportal-credentials secret")
+	admin_portal_url := "https://" + token3scale + "@" + admin_portal
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "adminportal-credentials",
+			Namespace: apicastNamespace,
+		},
+		Data: map[string][]byte{},
+	}
+	secret.Data["AdminPortalURL"] = []byte(admin_portal_url)
+	err := client.Create(ctx, secret)
+	if err != nil {
+		log.Error("Error create adminportal-credentials Secret in redhat-rhoam-operator namespace", err)
+		return err
+	}
+	return nil
+}
+
+// Use self-managed APIcast instead of the builded one for API
+// Update apicast secrets Hosts in redhat-rhoam-3scale
+// automates the manual steps in 3scale Admin Portal
+func updateThreeScaleRoutesToSelfManagedApicast(ctx context.Context, client k8sclient.Client) error {
+	log.Info("Update 3scale routes to selfmanaged-apicast-")
+	routes := routev1.RouteList{}
+	err := client.List(context.TODO(), &routes, &k8sclient.ListOptions{
+		Namespace: "redhat-rhoam-3scale",
+	})
+	if err != nil {
+		log.Error("Error obtaining routes list in redhat-rhoam-3scale namespace  ", err)
+		return err
+	}
+	for _, route := range routes.Items {
+		if strings.Contains(route.Spec.Host, "api-3scale-apicast-") {
+			log.Info("Route: " + route.Name + "  Host: " + route.Spec.Host)
+			routeSpecHost := route.Spec.Host
+			route.Spec.Host = strings.Replace(routeSpecHost, "api-3scale-apicast-", "selfmanaged-apicast-", 1)
+			log.Info("Host after replace: " + route.Spec.Host)
+			err := client.Update(ctx, &route)
+			if err != nil {
+				log.Error("error Update Route", err)
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// Install 3scale APIcast gateway operator
+func installThreeScaleApicastGatewayOperator(ctx context.Context, client k8sclient.Client, apicastNamespace string) error {
+	log.Info("Install 3scale APIcast gateway Operator")
+	targetNamespaces := []string{apicastNamespace}
+	og := &v1.OperatorGroup{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: apicastNamespace + "-",
+			Namespace:    apicastNamespace,
+			Generation:   1,
+			Annotations: map[string]string{
+				"olm.providedAPIs": "APIcast.v1alpha1.apps.3scale.net",
+			},
+		},
+		Spec: v1.OperatorGroupSpec{
+			TargetNamespaces: targetNamespaces,
+		},
+	}
+	err := client.Create(context.TODO(), og)
+	if err != nil && !k8serr.IsAlreadyExists(err) {
+		log.Error("Error create Operator Group in "+apicastNamespace+" namespace", err)
+		return err
+	}
+	subscription := &operatorsv1alpha1.Subscription{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "apicast-operator",
+			Namespace: apicastNamespace,
+			Labels: map[string]string{
+				"operators.coreos.com/apicast-operator.selfmanaged-apicast": "",
+			},
+			Generation: 1,
+		},
+		Spec: &operatorsv1alpha1.SubscriptionSpec{
+			Channel:                "threescale-2.10",
```

Comment:
The `2.10` is hard-coded here. Also, a few lines below `v0.4.0` is hard-coded. I think it is ok for now but I would print some warning message when running this script to update these once 3scale 2.11 is included in RHOAM.

Reply:
Thank you. I will do. TODO

Reply:
Done